src/main/java/com/aws/greengrass/clientdevices/auth/certificate/infra/BackgroundCertificateRefresh.java

```diff
@@ -171,7 +171,10 @@ private void scheduleNextRun() {
      */
     @Override
     public void accept(NetworkStateProvider.ConnectionState connectionState) {
-        if (connectionState == NetworkStateProvider.ConnectionState.NETWORK_UP) {
+        boolean shouldRun = isNetworkBackOnline();
+        this.previousConnectionState = connectionState;
+
+        if (shouldRun) {
```

Comment:
Tracking previous connection state doesn't seem right. It's an indirect mechanism which I think achieves the intended goal, but it makes the code harder to reason about and modify.

Can we just update `canRun()`?

```
    private boolean canRun() {
        // if nextScheduledRun is null then we have not started yet
        if (nextScheduledRun.get() == null) {
            return false;
        }
        Instant now = Instant.now();
        return now.equals(nextScheduledRun.get()) || now.isAfter(nextScheduledRun.get());
    }
```

Reply:
I was taking a look at your suggestion but going with that approach it will change a few things in terms of how the refresh works (which are minor). In the scenario we start the service and there is a network up and down in the first 24hours. We wouldn't trigger the refresh, given `nextScheduledRun` wouldn't be null but the seconds condition would return false

Reply:
That's the intended behavior, no?